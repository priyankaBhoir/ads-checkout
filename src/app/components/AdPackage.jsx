import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class AdPackage extends Component {
	static propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    description: PropTypes.array.isRequired
	}
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="icon-block" style={{height: 200}}>
				<h5 className="center">{this.props.title}</h5>
				<ul>{this.props.description.map((desc, index) => <li key={index}> {desc} </li>)} </ul>
				<div className="ad-price"> $ {this.props.price} </div>
			</div>
		)
	}
}
