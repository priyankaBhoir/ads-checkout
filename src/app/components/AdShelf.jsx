import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AdPackage from './AdPackage.jsx';

export default class AdShelf extends Component {
	static propTypes = {
	  ad: PropTypes.shape({
	    id: PropTypes.string.isRequired,
	    title: PropTypes.string.isRequired,
	    price: PropTypes.number.isRequired,
	    description: PropTypes.array.isRequired
	  }).isRequired,
	  addToCart: PropTypes.func.isRequired
	}
	constructor(props) {
		super(props);
		this.state = {
			quantity: 1
		}
	}

	updateQuntity = (e) => {
		this.setState({
			quantity: parseInt(e.target.value, 10)
		})
	} 

	addItem = () => {
		this.props.addToCart(this.props.ad, this.state.quantity);
	}

	render() {
		return (
			<div className="col s12 m4">
				<div> <AdPackage {...this.props.ad} /> </div>
				<div>	
					<input type="number" min="1" value={this.state.quantity} onChange={this.updateQuntity}/>
					<button onClick={this.addItem} > Add to Cart </button>
				</div>
			</div>
		)
	}
}
