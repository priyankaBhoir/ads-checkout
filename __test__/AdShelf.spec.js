import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import AdShelf from '../src/app/components/AdShelf'
import AdPackage from '../src/app/components/AdPackage'

// Snapshot for Home React Component
describe('AdShelf --- Snapshot',()=>{
    it('+++capturing Snapshot of AdShelf', () => {
        const ad = {
            id: 'classic',
            title: 'Classic',
            price: 299,
            description: ["Basic Ad"]
        };
        const renderedValue =  renderer.create(<AdShelf key={ad.id} ad={ad} addToCart={() => {}}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});

///*******************************************************************************************************
describe('Shallow Render REACT COMPONENTS :: AdShelf',()=>{
    let wrapper
    const ad = {
        id: 'classic',
        title: 'Classic',
        price: 299,
        description: ["Basic Ad"]
    };
    beforeEach(()=>{
        wrapper = shallow(<AdShelf key={ad.id} ad={ad} addToCart={() => {}}/>)
        
    })

    it('+++ render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('+++ contains header - AdPackage', () => {
        expect(wrapper.find(AdPackage).length).toEqual(1);
    });
    it('+++ contains a input initial value is 1 - AdPackage', () => {
        expect(wrapper.find('input').at(0).prop('value')).toEqual(1);
    });
    it('+++ Component calls addtoCart when buy is clicked', () => {
      const addtoCart = jest.fn();
      const wrapper = shallow(<AdShelf key={ad.id} ad={ad} addToCart={addtoCart}/>)

      const p = wrapper.find('button');
      p.simulate('click');
      expect(addtoCart).toBeCalledWith(ad, 1);
    });
});
