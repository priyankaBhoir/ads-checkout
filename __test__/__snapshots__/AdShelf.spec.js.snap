// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdShelf --- Snapshot +++capturing Snapshot of AdShelf 1`] = `
<div
  className="col s12 m4"
>
  <div>
     
    <div
      className="icon-block"
      style={
        Object {
          "height": 200,
        }
      }
    >
      <h5
        className="center"
      >
        Classic
      </h5>
      <ul>
        <li>
           
          Basic Ad
           
        </li>
         
      </ul>
      <div
        className="ad-price"
      >
         $ 
        299
         
      </div>
    </div>
     
  </div>
  <div>
    <input
      min="1"
      onChange={[Function]}
      type="number"
      value={1}
    />
    <button
      onClick={[Function]}
    >
       Add to Cart 
    </button>
  </div>
</div>
`;
